### DISCUSSION TITLE
How to get value of Select2 with crudfields js?

### DISCUSSION THREAD

**Discussion by fastdrums** [Laravel-Backpack/community-forum#386] [Q&A (Help)] [answered]
i have a select2_from_ajax field that gets its data from a fetch operation, and a 'custom html' element with an img tag.  I want to be able to change the image displayed depending on the item selected in the 'select2_from _ajax'.   The switch method for selects from the docs page doesnt seem to be working on select_2. 

How can I access the selected value in the select_2?

My code:
crud.field('landing_page_id').onChange(function(field) {

    switch(field.value) {
        case 1:
            source = "/images/landing_page/brand.png";
            break;
        case 2:
            source = "/images/landing_page/c_1.png";
            break;
        default:
            source = "/images/landing_page/empty.png";
    }

    $('img[id="lander-thumb"]').attr('src', source);

}).change();

**Comment by fastdrums** [comment, marked as answer]
i figured this out, leaving this here if anyone ever needs it.

to use with select_2, you have to put quotes around the values you are checking for.  
This works for me:

```javascript
crud.field('landing_page_id').onChange(function(field) {

switch(field.value) {
    case '1': // <---- here, it should be checked against string, not integer.
        source = "/images/landing_page/brand.png";
        break;
    case '2': // <---- here, it should be checked against string, not integer.
        source = "/images/landing_page/c_1.png";
        break;
    default:
        source = "/images/landing_page/empty.png";
}


$('img[id="lander-thumb"]').attr('src', source);
}).change();
```

**Reply by pxpm** [reply]
This is so weird, I didn't had time to check this deeply, but thanks for the workaround, I will try to reproduce this and come up with a solution if needed.

Thanks @fastdrums

**Reply by pxpm** [reply]
I get it now! Strings vs Ints. 👍 

I edited your comment to fix the code style display and add some comments, hope you don't mind 🙏 

I will be adding this bit of information to the docs too 👍 
EDIT: no need to add to docs, it's already there, from docs: `- `.value` - returns the value of that field (as a `string`);`

Thanks
